src/components/UsersDetailsCard.jsx

```diff
@@ -107,61 +122,66 @@ export default function UsersDetailsCard() {
   };
 
   const selectedIdArray = (id) => {
-    let arr = [...selectedArray];
+    let arr = [...selectedUsersIdArray];
     if (arr.includes(id)) {
-      setSelectedArray(arr.filter((ele) => ele !== id));
-      setSelectAll(false);
+      setState({ selectedUsersIdArray: arr.filter((ele) => ele !== id) });
+      setState({ selectAll: false });
     } else {
       arr.push(id);
-      setSelectedArray(arr);
-      if (arr.length === usersData.length) setSelectAll(true);
-      else setSelectAll(false);
+      setState({ selectedUsersIdArray: arr });
+      if (arr.length === usersData.length) setState({ selectAll: true });
+      else setState({ selectAll: false });
     }
   };
 
-  const deleteUserData = (selectedArray) => {
-    if (!selectedArray.length) return;
+  const deleteUserData = (selectedUsersIdArray) => {
+    if (!selectedUsersIdArray.length) return;
 
-    setSelectedArray([]);
-    setSelectAll(false);
-    setApiData(apiData.filter(({ id }) => !selectedArray.includes(id)));
+    setApiData(apiData.filter(({ id }) => !selectedUsersIdArray.includes(id)));
+    setState({ selectedUsersIdArray: [] });
+    setState({ selectAll: false });
   };
 
   const editUserDetails = (data) => {
     setApiData(apiData.map((ele) => (ele.id === data.id ? data : ele)));
-    handleClose();
+    setState({ openModal: false });
   };
 
   const selectAllFnc = () => {
-    setSelectedArray(usersData.map((userData) => userData.id));
-    setSelectAll(true);
+    setState({
+      selectedUsersIdArray: usersData.map((userdata) => userdata.id),
+    });
+    setState({ selectAll: true });
   };
 
   const deSelectAllFnc = () => {
-    setSelectedArray([]);
-    setSelectAll(false);
+    setState({ selectedUsersIdArray: [] });
+    setState({ selectAll: false });
   };
 
   const handleChange = (e) => {
-    setEditData({ ...editData, [e.target.name]: e.target.value });
+    setState({ editData: { ...editData, [e.target.name]: e.target.value } });
   };
 
   const handleOnClick = (button_id) => {
-    setSelectAll(false);
-    setSelectedArray([]);
-    setSelectedButton(button_id);
+    setState({ selectAll: false });
```

Comment:
No need to call setState 3 times, try understanding the reducer function logic. We are destructuring oldState & newState.
```
setState({ selectAll: false,  selectedUsersIdArray: [] ,  selectedButtonId: button_id });
```
Make this change wherever needed.

Reply:
done